File: app/src/main/java/remix/myplayer/service/Command.java
package remix.myplayer.service;

public interface Command {

  //控制命令
  int PLAYSELECTEDSONG = 0;
  int PREV = 1;
  int TOGGLE = 2;
  int NEXT = 3;
  int PAUSE = 4;
  int START = 5;
  int CHANGE_MODEL = 6;
  int LOVE = 7;
  int TOGGLE_MEDIASESSION = 8;
  int PLAY_TEMP = 9;
  int TOGGLE_NOTIFY = 10;
  int UNLOCK_DESKTOP_LYRIC = 11;
  int CLOSE_NOTIFY = 12;
  int ADD_TO_NEXT_SONG = 13;
  int CHANGE_LYRIC = 14;
  int PLAY_AT_BREAKPOINT = 15;
  int TOGGLE_TIMER = 16;
  int TOGGLE_DESKTOP_LYRIC = 17;
  int HEADSET_CHANGE = 18;
  int LOCK_DESKTOP_LYRIC = 19;
  int TOGGLE_STATUS_BAR_LRC = 20;
}

File: app/src/main/java/remix/myplayer/service/Command.kt
package remix.myplayer.service

interface Command {
  companion object {

    /**
     * 播放相关的操作才允许前台服务
     */
    fun isAllowForForegroundService(cmd: Int?) : Boolean{
      return cmd == PLAY_AT || cmd == SKIP_TO_PREVIOUS || cmd == SKIP_TO_NEXT || cmd == PLAY_PAUSE || cmd == PLAY_TEMP
    }

    // 控制命令
    const val PLAY_AT: Int = 0
    const val SKIP_TO_PREVIOUS: Int = 1
    const val PLAY_PAUSE: Int = 2
    const val SKIP_TO_NEXT: Int = 3
    const val PAUSE: Int = 4
    const val PLAY: Int = 5
    const val CHANGE_MODEL: Int = 6
    const val LOVE: Int = 7
    const val PLAY_TEMP: Int = 8
    const val UNLOCK_DESKTOP_LYRIC: Int = 9
    const val CLOSE_NOTIFY: Int = 10
    const val ADD_TO_NEXT_SONG: Int = 11
    const val TOGGLE_TIMER: Int = 12
    const val TOGGLE_DESKTOP_LYRIC: Int = 13
    const val TOGGLE_STATUS_BAR_LRC: Int = 14
    const val SEEK_TO: Int = 15
  }
}
